### ISSUE TITLE
Pope and wild animal players can be added twice

### ISSUE THREAD

**Issue by Nightinggale** [We-the-People-civ4col-mod/Mod#49]
In short: detection of existing players is based on names.

In most cases this is ok, but if somebody makes a scenario file, it can be problematic. The issue (which has happened a few times) is the scenario was made in **German** and when starting the game, the **English** names were not present and suddenly wild animals are present **both in German and English**. The same with the Pope.

Clearly the identification should not rely on the **_translated_** names.

**Comment by abmpicoli** [comment]
Since this only happens for scenarios, I think it is safe to say it should go to milestone 2.7.2...

**Comment by raystuttgart** [comment]
I never saw that.

And for all Scenario Maps we created in RaR this definitely never happened.
(We created them in a way that Wild Animals and Pope were added at game start by DLL)

But of course if you are not experienced in Map Making this could accidently happen to Players that create their own Scenario Maps.

**Comment by Nightinggale** [comment]
> But of course if you are not experienced in Map Making this could accidently happen to Players that create their own Scenario Maps.

People should be able to use WB to save a scenario and it should just work. Double up on wild animals shouldn't be possible regardless of how inexperienced the scenario creator is. I have no idea how common it is, but it has happened, which is enough to consider adding a flag to cleanly identify the special players regardless of language used by creator and player.

Adding a flag will also fix issues related to people renaming players, kind of like the folder name of the mod shouldn't be hardcoded because sometimes people have valid reasons to rename them, like having last stable and development at the same time. I can't think of a good reason to rename wild animals right now, but we shouldn't assume scenario creators would never come in a situation where it would make sense for the creator.

**Comment by Nightinggale** [comment]
I added code since this bug report was made, which could be helpful in fixing this. Rather than relying on names, it should ensure max one player of each of CIV_CATEGORY_BARBARIAN and CIV_CATEGORY_CHURCH.

```cpp
CvCivilizationInfo& CvPlayer::getCivilizationInfo() const;
CivCategoryTypes CvCivilizationInfo::getCivCategoryTypes() const;

enum CivCategoryTypes
{
	CIV_CATEGORY_EUROPEAN,
	CIV_CATEGORY_NATIVE,
	CIV_CATEGORY_KING,
	CIV_CATEGORY_BARBARIAN,
	CIV_CATEGORY_CHURCH,
	CIV_CATEGORY_NOT_SET,
};
```

**Comment by johan-buret** [comment]
I tested this with two WorldBuilders :

1. ``Americas Gigantic.ColonizationWBSave`` with the two empty slots at the end that got filled as usual
2. The savegame available below, where I reordered the players declaration, so that the Swedish King ends last.

Both detects properly the Church and Barbarian players
[DesertWarmer3.zip](https://github.com/We-the-People-civ4col-mod/Mod/files/13230715/DesertWarmer3.zip)
